### PR TITLE
[MM-28726] Add X to top right of Next Steps Tips for easy back-out

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
When in the Next Steps and Tips view, there was no easily noticeable way to back-out of it. This PR adds an X to the top right corner. When that X is clicked, it's the same behaviour as if the user clicked the X on the LHS - a modal appears stating that it will be added to the main menu.

Actually having it added to the main menu will come in a future PR, as part of ticket MM-28727 (which will build on this one)

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-28726

#### Related Pull Requests
n/a
#### Screenshots
image

### CODE REVIEW

**Comment by nickmisasi** [comment]
@matthewbirtch we were unsure of whether the X button on the tips and next steps screen should mimic the behaviour of the LHS x button, or just hide the tips and next steps screen (ie, it would come back on next page refresh, and still be available on the LHS). 

I went with the former, you can try it out in the spinwick. Let me know if you'd like it to be the latter.

**Comment by matthewbirtch** [comment]
@nickmisasi I think this works to behave consistently with the x in the LHS

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
There appears to be an issue with the highlighted area being off-position
image

Other than that, I think we're good

**Comment by nickmisasi** [comment]
@matthewbirtch highlighted area issue was because that spinwick was using the old sidebar, which isn't as wide. If you look now, it's good. I'll followup to make sure that the new sidebar is the only thing people will see in cloud.

**Review by stevemudie — APPROVED** [review]
LGTM

File: components/next_steps_view/next_steps_tips.tsx [hunk 1]
```diff
@@ -286,13 +335,20 @@ export default function NextStepsTips(props: { showFinalScreen: boolean; animati
...
+                />
             </header>
             <div className='NextStepsView__body'>
                 <div className='NextStepsView__nextStepsCards'>
                     <Card expanded={true}>
                         <div className='Card__body'>
-                            {// TODO: Bring back when the tour is working
-                            /* <h3>
+                            {
+
```
Comment (ali-farooq0) [inline components/next_steps_view/next_steps_tips.tsx:349]: ahhh eslint -- hate when it complains about not having an empty line above the comment :P

**Review by ali-farooq0 — APPROVED** [review]
👍

**Comment by agnivade** [comment]
My apologies for a completely unsolicited comment. But I got stuck in this step while signing up in https://hacktoberfest.cloud.mattermost.com/ and @ethervoid pointed me to this PR.

From what I understand, the big X button will just remove the tutorial. But there's no way to actually finish all the 3 steps in the progress bar in LHS. Which means there's _never_ a way for a user to get to see 3/3 steps completed, or something equivalent. This triggered my inner OCD and I kept helplessly clicking all the buttons possible, hoping for a way out of this challenge. :stuck_out_tongue_closed_eyes: 

As a user, I'll expect a Finish button of some sort to complete finish all the 3 steps after which the tutorial goes away, and if needed I can bring it back again from hamburger menu.

**Comment by matthewbirtch** [comment]
> From what I understand, the big X button will just remove the tutorial. But there's no way to actually finish all the 3 steps in the progress bar in LHS. Which means there's _never_ a way for a user to get to see 3/3 steps completed, or something equivalent. This triggered my inner OCD and I kept helplessly clicking all the buttons possible, hoping for a way out of this challenge. 😝

@agnivade you should be able to complete all 3 steps if you go through them and click the actions each stesp is asking you to do. If you're not able to successfully complete all 3 steps, then that would be another issue separate from what this ticket is addressing. If you're able to reproduce the errors, it would be helpful to either have a screen recording or the steps you took.

**Comment by nickmisasi** [comment]
@agnivade adding to Matt's comment, here's how your screen should look if you've interacted with all 3 steps on the first page: 
image

**Comment by agnivade** [comment]
Oh I know what the issue is. I _have_ to enable notifications, otherwise it never completes all the steps. Personally, I don't enable notifications from a website. But that's just my personal choice. IMO, it should be made clear that it's a compulsory step to complete the tutorial. I just thought that it was a recommendation. Sorry if it was obvious.

**Comment by matthewbirtch** [comment]
> Oh I know what the issue is. I _have_ to enable notifications, otherwise it never completes all the steps. Personally, I don't enable notifications from a website. But that's just my personal choice. IMO, it should be made clear that it's a compulsory step to complete the tutorial. I just thought that it was a recommendation. Sorry if it was obvious.

Makes sense @agnivade. We're making a nuanced update to this actually. You have to click the button, but then once the 'allow desktop notifications' confirm comes up, we will mark the step as complete even if you choose not to allow notifications

**Comment by marianunez** [comment]
> You have to click the button, but then once the 'allow desktop notifications' confirm comes up,

I see how someone may not even want to click "Allow Desktop Notifications" button at all given that they don't know they have an option to later click do not allow. It implies that it will be enabled as soon as it clicks. Should we have a secondary button there for Skip that step?

**Comment by nickmisasi** [comment]
> > You have to click the button, but then once the 'allow desktop notifications' confirm comes up,
> 
> I see how someone may not even want to click "Allow Desktop Notifications" button at all given that they don't know they have an option to later click do not allow. It implies that it will be enabled as soon as it clicks. Should we have a secondary button there for Skip that step?

The label for that button could be changed from "Enable Notifications" to something like "Set Notification Preferences", would show the user they'll have a choice

**Review by devinbinnie — APPROVED** [review]

**Comment by matthewbirtch** [comment]
> The label for that button could be changed from "Enable Notifications" to something like "Set Notification Preferences", would show the user they'll have a choice

Ya, that makes sense. Let's change it to 'Set up notifications'

**Review by matthewbirtch — APPROVED** [review]
> Let's change it to 'Set up notifications'

Approved with the change above

**Pull request merged**